pwiz_tools/Skyline/Util/ParsedMolecule.cs

```diff
@@ -204,6 +206,90 @@ public static bool TryParseFormula(string formula, out ParsedMolecule resultMole
             }
         }
 
+        /// <summary>
+        /// Separate the formula and mass offset declarations in a string presumed to describe a chemical formula and/or mass offsets.
+        /// Is aware of simple math, e.g. "C12H5[-1.2/1.21]-C2H[-1.1]" => "C12H5-C2H", 0.1, 0.11
+        /// </summary>
+        /// <param name="formulaAndMasses">input string  e.g. "C12H5", "C12H5[-1.2/1.21]", "[+1.3/1.31]", "1.3/1.31", "C12H5[-1.2/1.21]-C2H[-1.1]"</param>
+        /// <param name="formula">string with any mass modifiers stripped out e.g.  "C12H5[-1.2/1.21]-C2H[-1.1]" ->  "C12H5-C2H" </param>
+        /// <param name="modMassMono">effect of any mono mass modifiers e.g. "C12H5[-1.2/1.21]-C2H[-1.1]" => 0.1 </param>
+        /// <param name="modMassAverage">effect of any avg mass modifiers e.g. "C12H5[-1.2/1.21]-C2H[-1.1]" => 0.11</param>
+        private static void SplitFormulaAndMasses(string formulaAndMasses, out string formula, out double? modMassMono, out double? modMassAverage)
+        {
+            modMassMono = null;
+            modMassAverage = null;
+            formula = formulaAndMasses?.Trim();
+            if (string.IsNullOrEmpty(formula))
+            {
+                return;
```

Comment:
Should you normalize this to either "null" or String.Empty? As the code stands it will return either null or "".

I guess I would also mildly prefer this function to return a struct or class FormulaAndMasses object. Maybe Nick also said this.

Reply:
Then it would not have out parameters, and return a single object that you new at the top of the function.

Reply:
It's a local supporting function for something that does actually return a class, it feels strange to beat up on the heap for a thing like that.